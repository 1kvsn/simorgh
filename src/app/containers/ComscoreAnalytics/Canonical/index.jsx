/* eslint-disable jsx-a11y/alt-text */
import React, { useContext } from 'react';
import { Helmet } from 'react-helmet';
import { UserContext } from '#contexts/UserContext';

const CanonicalComscoreAnalytics = () => {
  const { cookiePolicy } = useContext(UserContext);

  let csUcfr = 0;
  if (cookiePolicy) {
    csUcfr = cookiePolicy.slice(-1);
  }

  return (
    <Helmet>
      <script
        async
        type="text/javascript"
        src={`${process.env.SIMORGH_PUBLIC_STATIC_ASSETS_ORIGIN}/simorgh/public/comscore.js`}
      />
      <noscript>
        {`<img src="https://sb.scorecardresearch.com/p?c1=2&c2=17986528&cs_ucfr=${csUcfr}&cv=2.0&cj=1" />`}
      </noscript>
    </Helmet>
  );
};

export default CanonicalComscoreAnalytics;
