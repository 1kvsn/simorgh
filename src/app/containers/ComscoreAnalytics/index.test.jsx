import React from 'react';
import { render } from '@testing-library/react';
import ContextWrap from './utilities/testHelper';
import ComscoreAnalytics from '.';

describe('Comscore Analytics Container', () => {
  describe('Assertions - AMP', () => {
    it('should return null when toggle is disabled', () => {
      const toggleState = {
        comscoreAnalytics: {
          enabled: false,
        },
      };
      const { container } = render(
        <ContextWrap
          platform="amp"
          pageType="article"
          origin="bbc.com"
          toggleState={toggleState}
        >
          <ComscoreAnalytics />
        </ContextWrap>,
      );

      expect(container).toBeEmptyDOMElement();
    });
  });

  describe('Assertions - Canonical', () => {
    it('should return null when toggle is disabled', async () => {
      const toggleState = {
        comscoreAnalytics: {
          enabled: false,
        },
      };
      const { container } = render(
        <ContextWrap
          platform="canonical"
          pageType="article"
          origin="bbc.com"
          toggleState={toggleState}
        >
          <ComscoreAnalytics />
        </ContextWrap>,
      );

      expect(container).toBeEmptyDOMElement();
    });
  });
});
