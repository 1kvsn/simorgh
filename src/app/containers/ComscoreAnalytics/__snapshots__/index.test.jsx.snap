// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comscore Analytics Container Snapshots should render comscore script when on amp and toggle is enabled 1`] = `
<amp-analytics
  type="comscore"
>
  <script
    type="application/json"
  >
    {"vars":{"c2":"17986528"},"extraUrlParams":{"comscorekw":"amp"}}
  </script>
</amp-analytics>
`;

exports[`Comscore Analytics Container Snapshots should render comscore script when on canonical and toggle is enabled 1`] = `
<html>
  <head>
    <noscript>
      &lt;img src="https://sb.scorecardresearch.com/p?c1=2&c2=17986528&cs_ucfr=0&cv=2.0&cj=1" /&gt;
    </noscript>
    <script
      async="true"
      src="http://localhost:7080/comscore.js"
      type="text/javascript"
    />
  </head>
  <body>
    <div />
  </body>
</html>
`;
